subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -423,16 +423,37 @@ int fota_download(const char *host, const char *file,
 	const int *sec_tag_list, uint8_t sec_tag_count, uint8_t pdn_id, size_t fragment_size,
 	const enum dfu_target_image_type expected_type)
 {
-	uint32_t host_hash = 0;
-	uint32_t file_hash = 0;
-	int err = -1;
 	static int sec_tag_list_copy[CONFIG_FOTA_DOWNLOAD_SEC_TAG_LIST_SIZE_MAX];
-
 	struct download_client_cfg config = {
 		.pdn_id = pdn_id,
 		.frag_size_override = fragment_size,
 	};
 
+	if (sec_tag_count > ARRAY_SIZE(sec_tag_list_copy)) {
+		return -E2BIG;
+	}
+
+	if ((sec_tag_list != NULL) && (sec_tag_count > 0)) {
+		memcpy(sec_tag_list_copy, sec_tag_list, sec_tag_count * sizeof(sec_tag_list[0]));
+
+		config.sec_tag_count = sec_tag_count;
+		config.sec_tag_list = sec_tag_list_copy;
+
+		if (!is_ip_address(host)) {
+			config.set_tls_hostname = true;
+		}
+	}
+
+	return fota_download_v2(host, file, &config, expected_type);
```

Comment:
We have a bug here.

sec_tag_list_copy is overwritten without checking if that is already in use.
Why exactly do we copy, when the `fota_download_cfg()` does not copy it? The behaviour here is different.

And because we don't copy URI, I recommend that we do the same for sec_tag list. Otherwise the API is a bit mixed up because it has two different memory handling strategies.